lib/api/core/models/repositories/userRepository.js

```diff
@@ -126,8 +128,64 @@ class UserRepository extends Repository {
             return user;
           });
       });
+  }
 
+  /**
+   * Given a User object :
+   * - delete it from memory and database
+   * - delete related token
+   * - delete related stategies
+   *
+   * @param {User} user
+   * @param {object} [options]
+   * @returns {Promise}
+   */
+  delete (user, options = {}) {
+    return super.delete(user, options)
+      .then(() => this.kuzzle.repositories.token.deleteByUserId(user._id))
+      .then(() => removeUserStrategies(this.kuzzle, user))
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Move the declaration of "removeUserStrategies" before this usage. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)